File: Python/ejercicio04.py
"""
* Escribe un programa que se encargue de comprobar si un número es o no primo.
* Hecho esto, imprime los números primos entre 1 y 100.
"""
# recorremos los num del 2 al 100
for i in range(2, 101):

    #asumimos que el numero es primo
    esPrimo = True

    #comprobamos divisores
    for j in range(2, i):

        #si la division es excata
        if i % j == 0:

            #entonces NO es primo
            esPrimo = False

    #si sigue siendo primo lo imprimimos
    if esPrimo:
        print(i)

File: Java/Ejercicio04.java
/*
 * Escribe un programa que se encargue de comprobar si un número es o no primo.
 * Hecho esto, imprime los números primos entre 1 y 100.
 */

public class Ejercicio04 {

    public static void main(String[] args) {

        // recorremos los num del 2 al 100
        for (int i = 2; i <= 100; i++) {

            // asumimos que el num es primo
            boolean esPrimo = true;

            // comprobamos divisores
            for (int j = 2; j < i; j++) {

                // si la division es exacta
                if (i % j == 0) {

                    // entonces NO es primo
                    esPrimo = false;
                }
            }
            // si sigue siendo primo lo imprimimos
            if (esPrimo) {
                System.out.println(i);
            }

        }
    }

}
